console-subscriber/src/aggregator/mod.rs

```diff
@@ -770,49 +784,98 @@ impl Aggregator {
             }
 
             Event::StateUpdate {
-                resource_id,
+                update_id,
+                update_type,
                 update,
                 ..
             } => {
-                let resource_id = self.ids.id_for(resource_id);
-                if let Some(mut stats) = self.resource_stats.update(&resource_id) {
-                    let field_name = match update.field.name.clone() {
-                        Some(name) => name,
+                let update_id = self.ids.id_for(update_id);
+                let mut to_update = vec![(update_id, update_type.clone())];
+
+                fn update_entry(e: Entry<'_, FieldKey, Attribute>, upd: &AttributeUpdate) {
+                    e.and_modify(|attr| update_attribute(attr, upd))
+                        .or_insert_with(|| upd.clone().into());
+                }
+
+                match update_type {
+                    UpdateType::Resource => {
+                        if let Some(parent) = self
+                            .resources
+                            .get(&update_id)
+                            .and_then(|r| self.resources.get(r.parent_id.as_ref()?))
+                            .filter(|parent| parent.inherit_child_attrs)
+                        {
+                            to_update.push((parent.id, UpdateType::Resource));
+                        }
+                    }
+                    UpdateType::AsyncOp => {
+                        if let Some(parent) = self
+                            .async_ops
+                            .get(&update_id)
+                            .and_then(|r| self.async_ops.get(r.parent_id.as_ref()?))
+                            .filter(|parent| parent.inherit_child_attrs)
+                        {
+                            to_update.push((parent.id, UpdateType::AsyncOp));
+                        }
+                    }
```

Comment:
Alternatively, you could have a method that takes an `UpdateType` and gives you back the hashmap, and then leave the currently duplicated code inline as a single non-duplicated `if` statement. (This works in part because you can just pass along the `UpdateType` into the `to_update.push` invocation, rather than writing out the specific variant the way you are doing here.

Reply:
yeah, that's more or less what i was imagining. i think the `filter` might have to be kept separate, as well, because the field access is being performed on a different type. we could introduce a trait for that, but, hmm...